dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx

```diff
@@ -74,6 +78,20 @@ export const TopicFilterBank = ({
 	filterKeyEvents = false,
 }: Props) => {
 	const palette = decidePalette(format);
+
+	const selectedTopic = selectedTopics?.[0];
+	const topFiveTopics = availableTopics.slice(0, 5);
+
+	if (selectedTopic) {
+		const selectedIndex = availableTopics.findIndex((availableTopic) =>
+			isEqual(selectedTopic, availableTopic),
+		);
+
+		if (selectedIndex > 4) {
+			topFiveTopics.splice(4, 1, availableTopics[selectedIndex]);
+		}
+	}
```

Comment:
This code is not following the React lifecycle, and should probably use hooks rather than splicing data!

Reply:
Oh really? thanks for letting us know. I'll create a ticket in our board to look into it 👍

Reply:
If it works, I guess it’s not a major necessity, but generally, you would use a useEffect hook to compute this value.

However, in this case, I’m wondering if it would be beneficial to simply create the the array in one pass, which would 

```ts
const relevantTopics = availableTopics.filter(({ count }) => (count ?? 0) > 2);
/** take all the selected topics,
 * append the relevant remaining ones
 * and keep the first five
 */
const topFiveTopics = selectedTopics.concat(relevantTopics).slice(0, 5);
```

Reply:
Here’s the React docs on mutating data: https://react.dev/learn/updating-arrays-in-state#updating-arrays-without-mutation

Reply:
Wrt to using hooks I think this looks okay as the state is just derived state. Afaics the state is not React owned state managed by useState or a useEffect dependency with the expectation of updates that will trigger a re-render.

I think the docs are aimed at when arrays are used as React state and are being mutated in place which is 'bad' as mutations then will not trigger re-renders.

Agree however we should not mutate the array in place to avoid any side effects in JS or React if the state is used differently in future.